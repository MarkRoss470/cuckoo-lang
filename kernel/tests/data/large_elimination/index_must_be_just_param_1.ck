data Bool : Type where
  | true : Bool
  | false : Bool

def Bool.or (a : Bool) (b : Bool) : Bool := Bool.rec.{1}
    (fun (_ : Bool) => Bool)
    Bool.true
    b
    a

data SomeProp : Bool -> Prop where
  | c : (a : Bool) -> (b : Bool) -> SomeProp (Bool.or a b)

-- Both `a` and `b` are mentioned in the index of `SomeProp.c`, but `SomeProp` is still not large eliminating
-- because they are not used verbatim as an index
def some_prop_rec : Prop := SomeProp.rec.{1}