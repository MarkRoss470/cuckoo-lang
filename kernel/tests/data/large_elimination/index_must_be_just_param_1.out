ERROR at tests/data/large_elimination/index_must_be_just_param_1.ck 16:29-16:45: SomeProp.rec takes 0 level argument(s), but 1 were provided.